modyn/trainer_server/internal/trainer/remote_downsamplers/abstract_remote_downsample_strategy.py

```diff
@@ -5,39 +5,49 @@
 
 
 def get_tensors_subset(
-    indexes: torch.Tensor, data: Union[torch.Tensor, dict], target: torch.Tensor, sample_ids: list
-) -> tuple[Union[torch.Tensor, dict], torch.Tensor, list]:
+    selected_indexes: list[int], data: Union[torch.Tensor, dict], target: torch.Tensor, sample_ids: list
+) -> tuple[Union[torch.Tensor, dict], torch.Tensor]:
+    # Starting from the batch, we want to keep only the selected samples
+
+    # first of all we compute the position of each selected index within the batch
+    in_batch_index = [sample_ids.index(selected_index) for selected_index in selected_indexes]
+
+    # then we extract the data
     if isinstance(data, torch.Tensor):
-        sub_data = data[indexes]
+        sub_data = data[in_batch_index]
     elif isinstance(data, dict):
-        sub_data = {key: tensor[indexes] for key, tensor in data.items()}
+        sub_data = {key: tensor[in_batch_index] for key, tensor in data.items()}
 
-    sub_target = target[indexes]
-    sub_sample_ids = [sample_ids[i] for i in indexes]
+    # and the targets
+    sub_target = target[in_batch_index]
 
-    return sub_data, sub_target, sub_sample_ids
+    return sub_data, sub_target
 
 
 class AbstractRemoteDownsamplingStrategy(ABC):
-    def __init__(self, params_from_selector: dict) -> None:
-        assert "downsampled_batch_size" in params_from_selector
-        self.downsampled_batch_size = params_from_selector["downsampled_batch_size"]
+    def __init__(self, pipeline_id: int, trigger_id: int, batch_size: int, params_from_selector: dict) -> None:
+        self.pipeline_id = pipeline_id
+        self.batch_size = batch_size
+        self.trigger_id = trigger_id
+
+        assert "downsampled_batch_ratio" in params_from_selector
+        self.downsampled_batch_ratio = params_from_selector["downsampled_batch_ratio"]
```

Comment:
yeah this is I think a problem for BtS right now where it's always 1.0... but not sure